synthesizer/program/src/lib.rs

```diff
@@ -983,6 +983,22 @@ impl<N: Network> ProgramCore<N> {
         })
     }
 
+    /// Returns `true` if a program contains any V13 syntax.
+    /// This includes `Operand::Generator`.
+    /// This is enforced to be `false` for programs before `ConsensusVersion::V13`.
+    #[inline]
+    pub fn contains_v13_syntax(&self) -> bool {
+        // Check each instruction and output in each function's finalize scope for the use of
+        // `Operand::Generator`.
+        cfg_iter!(self.functions()).any(|(_, function)| {
+            function.finalize_logic().is_some_and(|finalize_logic| {
+                cfg_iter!(finalize_logic.commands()).any(|command| {
+                    cfg_iter!(command.operands()).any(|operand| matches!(operand, Operand::Generator(_)))
```

Comment:
Nested parallel iteration may degrade performance

Reply:
I kept it consistent with the checks from previous consensus versions. 

In the future we can update this code to perform a singular scan of these objects rather than doing it once per consensus version change.